core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Parameter.kt

```diff
@@ -3,22 +3,26 @@ package ch.ergon.dope.resolvable.expression.unaliased.type
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.validtype.ValidType
 
-class Parameter<T : ValidType>(val previousValue: TypeExpression<T>, private var parameterName: String?) : TypeExpression<T> {
+class ValuedParameter<T : ValidType>(val value: TypeExpression<T>, private val parameterName: String?) : TypeExpression<T> {
```

Comment:
Can't we make parameterName non-nullable, and pass the parameter-count as a default if no name is provided?